product-service/src/lambda/catalog-batch-process.ts

```diff
@@ -0,0 +1,61 @@
+import { APIGatewayProxyHandler } from "aws-lambda";
+import AWS from "aws-sdk";
+
+import { httpResponse } from "../libs/http";
+import { lambda } from "../libs/lambda";
+import { productService } from "../services/product-service";
+
+const PRODUCT_TOPIC_ARN = process.env.PRODUCT_TOPIC_ARN;
+
+const withParams = product => ({
+  Message: JSON.stringify(product),
+  TopicArn: PRODUCT_TOPIC_ARN,
+  Subject: `Hey! Somebody has uploaded files`

```

Comment:
It seems that filtering wouldn't work with this params. You have no message attributes. https://docs.aws.amazon.com/sns/latest/dg/sns-message-attributes.html

Reply:
Thank you a lot, fix it